src/libhirte/common/cfg.c

```diff
@@ -133,6 +138,96 @@ int cfg_load_from_file(struct config *config, const char *config_file) {
+
+                while (current_node != NULL) { // search for the correct place to insert
+                        if (strcmp(file_location, current_node->file_location) < 0) {
+                                break;
+                        }
+                        previus_node = current_node;
+                        current_node = current_node->next;
+                }
+                struct node *new_node = (struct node *) malloc(sizeof(struct node));
+                new_node->file_location = file_location;
+                if (current_node == NULL) { // means it's suppose to be last
+                        new_node->next = NULL;
+                        previus_node->next = new_node;
+                } else {                                 // insert between current node and previous one
+                        if (current_node == list_head) { // means it suppose to be first
+                                new_node->next = list_head;
+                                list_head = new_node;
+                        } else {
+                                new_node->next = current_node;
+                                previus_node->next = new_node;
+                        }
+                }
+        }
+}
+
+void cfg_load_files_from_list(struct config *config) {
+        struct node *tmp = list_head;
+
+        if (list_head == NULL) { // nothing to load
+                return;
+        }
+
+        while (tmp != NULL) {
+                cfg_load_from_file(config, tmp->file_location);
+                tmp = tmp->next;
+        }
+}
+
+void free_list() {
+        struct node *current = list_head;
+        struct node *prev = NULL;
+
+        while (current != NULL) {
+                prev = current;
+                current = current->next;
+                free(prev);
+        }
+}
+
+char *is_file_name_ending_with_conf(char *file_name, char *conf_file_enfing) {
+        return strstr(file_name, conf_file_enfing);
+}
+
+void reconstract_file_and_add_to_list(const char *custom_config_directory, char *file_name) {
+        int size_of_full_file_location = (int) (strlen(custom_config_directory) + strlen(file_name) + 1);
+        char *file_full_path = strdup(custom_config_directory);         // add the path
+        strncat(file_full_path, "/", size_of_full_file_location);       // add '/'
+        strncat(file_full_path, file_name, size_of_full_file_location); // add file name
+        add_file_to_list_and_sort(file_full_path);
+}
+
+int cfg_load_from_dir(struct config *config, const char *custom_config_directory) {
+        list_head = NULL;                 // start new head
+        char *conf_file_enfing = ".conf"; // needed ending for configuration files
+        DIR *dir = opendir(custom_config_directory);
+        if (!dir) {
+                return ENOENT;
+        }
+        struct dirent *current_file = NULL;
+        while ((current_file = readdir(dir)) != NULL) { // while there is files
```

Comment:
It would probably be easier to use scandir function, which could sort filename already, examples below

https://lloydrochester.com/post/c/list-directory/
https://stackoverflow.com/questions/18402428/how-to-properly-use-scandir-in-c